### ISSUE TITLE
Information: recent omniORB CMake port working well for MinGW64

### ISSUE THREAD

**Issue by spl1inter** [msys2/MINGW-packages#7334]
Hi. For anyone needing OmniORB/CORBA/etc under MinGW64 have a look at the recent CMake port by Julian Scholle:
[https://github.com/CaeruleusAqua/omniORB-cmake](https://github.com/CaeruleusAqua/omniORB-cmake)

Seems working OOTB, only minor tweaks needed.

**Comment by lazka** [comment]
Just wondering, who/what is still using CORBA?

**Comment by revelator** [comment]
Hmm firebird2 is kind of corba (based on the old borland database), Orbit-2 is also still used on linux for exchange mail clients.
Actually i managed to build a working version of Orbit-2 and bonobo but the build steps are kind of akward as mingw Orbit-2 relies on having a working Orbit-2 as a bootstrapper in Msys2. So we could have a windows port of later versions of the evolution mail client again shortly.

**Comment by spl1inter** [comment]
AFAIK It's still a build requirement for Salome platform:
https://www.salome-platform.org/

In any case, there are hundreds of legacy CORBA apps, so having a recent ORB compatible that's (still) easy enough to build and available probably isn't a bad idea.

**Comment by revelator** [comment]
Personally newer had much luck compiling omniorb but if you can then id suggest making a PKGBUILD with the right flags patches etc. :)

**Comment by spl1inter** [comment]
Exactly what I said :) I thought about doing a PKGBUILD that but there are a couple of issues. First I am not the actual maintainer and I am not sure that person even works on it anymore. Second is that I use github very rarely myself. My advice for anyone who would have use for OmniORB simply clone and build hist project directly -> it actually does work and the binaries are the most reliable that I have had for MinGW64. My own intentions don't really go much further than that. I wish I had time to be more active with this project.

**Comment by revelator** [comment]
hmm well my first try ended miserably and there does not seem to be much info on anything specific that needs to be set on his page. Ill see if i can dig anything further up on it but if you know the build setup it might be better to post it here so i dont have to jump through hoofs to get it running.

**Comment by spl1inter** [comment]
I used cmake-gui, being sure to build in a subdir. I am attaching my CMakeCache

[CMakeCache.txt](https://github.com/msys2/MINGW-packages/files/5629059/CMakeCache.txt)

**Comment by revelator** [comment]
Hmm i used python3 since python2 is deprecated but else pretty much the same, though i did build in bash with mingw64 cmake so the path mangling might act up there ?, atleast im having no joy getting it to compile that way which might indicate a problem as that would rule out PKGBUILD since that relies on bash. Ill try with the override to the msys2 path mangling and see what happens.

**Comment by spl1inter** [comment]
Where is the build breaking? Maybe I can assist..

**Comment by revelator** [comment]
`/bin/sh: C:/Msys64/src/64/bin: Is a directory
make[2]: *** [CMakeFiles/RunGenerator.dir/build.make:129: generated/lib/omniORB/omniORB4/Naming.hh] Error 126
make[1]: *** [CMakeFiles/Makefile2:698: CMakeFiles/RunGenerator.dir/all] Error 2
make: *** [Makefile:149: all] Error 2`

**Comment by revelator** [comment]
Normally that would also be cool but if you want it added as a package we need to find a solution so that it will work with pacman :)

**Comment by spl1inter** [comment]
Agreed, I'm jacked in now give me a few I'll update with my next questions..

**Comment by revelator** [comment]
Np ;)

**Comment by revelator** [comment]
Ill zip em up for you tomorrow, and yeah seems like the generator does not like msys2's path mangling i suspect it should have run the just built omniidl.exe to generate the interface header. 
This is btw why it is somewhat akward to build Orbit2-idl also as it needs an Msys2 version of the idl compiler to get around the path mangling in the mingw* builds. I found that out from the cygwin version so there does not seem to be any easy solution.

**Comment by spl1inter** [comment]
Sure. And yeah that would explain it. The way I have things set up is almost default with just one tweak:
* pacman installed MSYS2 and mingw64 at a top level '/MSYS2' directory
* use '/MSYS2/mingw64' as the installation prefix for all my builds (note the absence of a drive letter)
* the tricky bit: a simple NTFS symbolic link from /MSYS2 to /MSYS2/MSYS2 - this is because MinGW64 will use actual drive locations which MSYS cannot understand when its something inside its own file structure ('/MSYS2/bin/blah'.. becomes '/bin/blah' so a MinGW build script becomes gibberish to MSYS).  This simple symbolic link was key to smoothly building complex packages like TeX Live which can't understand the constant switch in 'name-mangling' between MinGW64 and MSYS2 tools. I think you get the idea.. maybe it will help in your case..

That's about it. You have my cmakecache and the MinGW built omniidl completeed the rest of the build just fine -> no need for an MSYS version AFAICS

**Comment by revelator** [comment]
Yep this also plagued the older Msys as paths just ended in nothing if the code couldnt handle the windows path conversion.
Could probably use your trick to make a PKGBUILD though it would look somewhat odd to the uninformed :)

**Comment by revelator** [comment]
Or maybe use shell escaping ? hmm.

**Comment by spl1inter** [comment]
>  this also plagued the older Msys as paths just ended in nothing if the code couldnt handle the windows path conversion

But it's worse in MSYS2 for two reasons:
* MSYS2 uses a stricter fakeroot
* MinGW64 is installed as a subset of MSYS2, the old mingw32 this was reverse, so MSYS and MinGW could both use /MinGW/MSYS -> soemthing no longer possible

It annoyed me for a while until I finally gave up and put in the symbolic link, never looked back since.
BTW I wrote that wrong having MSYS on the brain too much, for correctness:
```
D:\MSYS2>dir /a
...
01/10/2020  07:37 PM    <SYMLINKD>     MSYS2 [\MSYS2]
...
```
IMHO shell escaping is always a dogs breakfast, I only need to escape the quotes in one of the cmake file that was it.

**Comment by revelator** [comment]
Hehe i know the feeling :) been bashing my head in sometimes trying to workaround these cases.
Atleast we have a working solution though When i used the old Msys it was /Msys/mingw though you could do it the other way around to.

**Comment by spl1inter** [comment]
> When i used the old Msys it was /Msys/mingw though you could do it the other way around to.

Really? I'm not sure. I just went with whatever the packaging system did, at that time it was mingw-get. All of the msys stuff went into /MinGW/msys/1.0 or something like that. Later mingw32 adopted msys2 and put that in a subdir as well. I didn't try to change much because it would raise additional issues when asking for support.

_"bash-ing my head"_ lol thats a good one.

Let me know how you go.

**Comment by revelator** [comment]
Was way back just about when the old Msys project started using mingw-get, i even did some work for them back then trying to port the newer cygwin into a new Msys2 (not related to this one) with some success though at the time i was not yet very good with this build system so it newer really took off. 
I did notice some packagers used the /mingw/msys layout later though.

Atm. im getting ready for some of my friends who are comming to dinner but i might have an idea for a PKGBUILD script ;)
